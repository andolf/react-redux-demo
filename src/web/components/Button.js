import React from 'react';
import classNames from 'classnames';
import style from '../style/components/Button.css';

const Button = ({type, className = '', children}) => (
    <button type={type} className={classNames(style.button, className)}>
        {children}
    </button>
);

export default Button;
