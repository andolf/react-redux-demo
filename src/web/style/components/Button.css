@import "../core/variables.css";

.button {
  color: var(--blue-dark);
  font-family: "Arial", sans-serif;
}
