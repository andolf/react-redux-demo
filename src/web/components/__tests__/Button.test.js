import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button';

test('Button renders with default type', () => {
    const component = renderer.create(
        <Button>Text</Button>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
