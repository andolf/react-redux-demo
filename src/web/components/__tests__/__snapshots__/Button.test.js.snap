// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders with default type 1`] = `
<button
  className=""
  type="button"
>
  Text
</button>
`;
